src/VideoUtils.ts

```diff
@@ -32,18 +49,42 @@ function durationToTotalChunks(duration: string): number {
 export async function getVideoInfo(videoGuids: Array<string>, session: Session, subtitles?: boolean): Promise<Array<Video>> {
     let metadata: Array<Video> = [];
     let title: string;
-    let date: string;
+    let duration: string;
+    let publishDate: string;
+    let publishTime: string;
+    let author: string;
+    let authorEmail: string;
+    let uniqueId: string;
+    const outPath = '';
     let totalChunks: number;
     let playbackUrl: string;
     let posterImageUrl: string;
     let captionsUrl: string | undefined;
 
     const apiClient: ApiClient = ApiClient.getInstance(session);
 
-    for (const GUID of videoGuids) {
-        let response: AxiosResponse<any> | undefined= await apiClient.callApi('videos/' + GUID, 'get');
+    /* TODO: change this to a single guid at a time to ease our footprint on the
+    MSS servers or we get throttled after 10 sequential reqs */
+    for (const guid of videoGuids) {
+        let response: AxiosResponse<any> | undefined =
+            await apiClient.callApi('videos/' + guid + '?$expand=creator', 'get');
+
+        title = sanitizeWindowsName(response?.data['name']);
+
+        duration = isoDurationToString(response?.data.media['duration']);
+
+        publishDate = publishedDateToString(response?.data['publishedDate']);
+
+        publishTime = publishedTimeToString(response?.data['publishedDate']);
+
+        author = response?.data['creator'].name;
+
+        authorEmail = response?.data['creator'].mail;
+
+        uniqueId = '#' + crypto.createHash('sha1').update(guid).digest('hex').substring(0, 8);
```

Comment:
Maybe simply extract the first group from the video's GUID instead? That's going to be unique _enough_ for our purposes while also allowing the users to still reconcile the source URL with the resulting file name on disk.

Reply:
Why not also hashing? The cost of a sha-1 it's so negligible

Reply:
great

Reply:
✅